src/diffusers/models/unets/unet_stable_cascade.py

```diff
@@ -21,9 +22,94 @@
 import torch.nn as nn
 
 from ...configuration_utils import ConfigMixin, register_to_config
-from ...utils import BaseOutput
+from ...loaders.single_file_utils import load_single_file_model_checkpoint
+from ...utils import BaseOutput, is_accelerate_available
 from ..attention_processor import Attention
-from ..modeling_utils import ModelMixin
+from ..modeling_utils import ModelMixin, load_model_dict_into_meta
+
+
+if is_accelerate_available():
+    from accelerate import init_empty_weights
+
+
+DEFAULT_CONFIGS = {
+    "stage_c": {"pretrained_model_name_or_path": "diffusers/stable-cascade-configs", "subfolder": "prior"},
+    "stage_c_lite": {"pretrained_model_name_or_path": "diffusers/stable-cascade-configs", "subfolder": "prior_lite"},
+    "stage_b": {"pretrained_model_name_or_path": "diffusers/stable-cascade-configs", "subfolder": "decoder"},
+    "stage_b_lite": {"pretrained_model_name_or_path": "diffusers/stable-cascade-configs", "subfolder": "decoder_lite"},
+}
+
+
+def convert_single_file_to_diffusers(original_state_dict):
```

Comment:
We could leverage these functions in the conversion script to massively reduce the duplicated code, no? 

(Future PR candidate)

Reply:
I copied these over from the conversion script. But yes, I can update the scripts to use this function once merged.